main.go

```diff
@@ -252,6 +270,28 @@ func waitForRayClusterAPIandSetupController(ctx context.Context, mgr ctrl.Manage
 	}
 }
 
+func setupAppWrapperController(mgr ctrl.Manager, cfg *config.CodeFlareOperatorConfiguration, certsReady chan struct{}) {
+	setupLog.Info("Waiting for certificate generation to complete")
+	<-certsReady
+	setupLog.Info("Certs ready")
+
+	if cfg.AppWrapper != nil && ptr.Deref(cfg.AppWrapper.Enabled, false) {
+		exitOnError(awctrl.SetupWebhooks(mgr, cfg.AppWrapper.Config), "error setting up AppWrapper webhook")
+		exitOnError(awctrl.SetupControllers(mgr, cfg.AppWrapper.Config), "error setting up AppWrapper controller")
+	} else {
+		setupLog.Info("AppWrapper controller disabled", "Config flag value", *cfg.AppWrapper.Enabled)
+	}
+}
+
+func setupIndexers(ctx context.Context, mgr ctrl.Manager, cfg *config.CodeFlareOperatorConfiguration) error {
```

Comment:
This creates a hard dependency on the Kueue CRDs being installed in the cluster before this operator starts up when the AppWrapper controller is enabled. Unfortunately, unlike controllers and web hooks, we have to setup the indexers before the controller-runtime's manager starts so we can't adopt the pattern you use in the RayController of waiting until the CRDs are available to enable the controller.

Reply:
Could exiting the main process in the watch loop be a solution? That is watching for the AppWrapper CRD to be established, then exiting the process with return code 0, let the Pod restarts, and let it run the branch where the API is installed and setup the indexer and the controller.

Reply:
That could work.   I'll see if I can generalize the logic in `waitForRayClusterAPIandSetupController` so it can parameterized by the CRDs to watch for and the action to take when they become available.